Formula/lighthouse.rb

```diff
@@ -0,0 +1,35 @@
+class Lighthouse < Formula
+  desc "Rust Ethereum 2.0 Client"
+  homepage "https://github.com/sigp/lighthouse"
+
+  url "https://github.com/sigp/lighthouse.git",
+      tag:      "v1.4.0",
+      revision: "3b600acdc5bf9726367c18277a22486573b8b457"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+  depends_on "rustup-init" => :build
+
+  # Update the Rust toolchain version whenever lighthouse is updated.
+  def rust_toolchain
+    "1.52.1"
+  end
+
+  # Update the version string whenever lighthouse is updated.
+  def lighthouse_version_string
+    "v1.4.0-3b600ac"
+  end
+
+  def install
+    # This will install a rust toolchain to be used with lighthouse.
+    system Formula["rustup-init"].bin/"rustup-init", "-qy", "--no-modify-path",
+           "--default-toolchain", rust_toolchain, "--profile", "minimal"
+    ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
```

Comment:
Why not depend on homebrew's rust?

Reply:
Unfortunately due to the breaking changes in the most recent version of the rust compiler the latest version of `lighthouse` does not compile with it. 

Hopefully, the next release `1.5.0` will solve that. Then I'll update the whole formula to use the homebrew rust.

Reply:
We'll just wait with this PR until the next version then. The current solution is bad for maintainability.

Reply:
Asking out of curiosity: Rust compiler move fast and break things often. Formulae that compiles now may suddenly stop working in a few months' time for someone. How I can protect my formulae from such a scenario?

Reply:
Make sure that the project you're following actually keeps up to date with the latest changes in the language, maybe ask them to run CI on both a stable and nightly version.

Reply:
We distribute pre-compiled binaries, so users on a supported OS will continue to be able to `brew install foo` even after `foo` fails to compile with Rust.

That said, there are ~200 formulae that build with Rust, and only a handful of them get broken on any given Rust version update, and those that do are almost always fixed when they are updated.